cache.go

```diff
@@ -0,0 +1,192 @@
+package libovsdb
+
+import (
+	"reflect"
+	"sync"
+)
+
+// RowCache is a collections of Rows hashed by UUID
+type RowCache struct {
+	cache map[string]Row
```

Comment:
A thought on this: if we want to add the ORM stuff on top of this we have two ways to go:
**A** - Decode on DB updates and store the decoded objects on the cache (similar to what I proposed in the [go-ovn PR](https://github.com/eBay/go-ovn/blob/15eca9a4c0d78063b44b9cb844ab1c2d3347a226/client.go#L313))
**B** - Decode the objects on API calls

Though B) might be more efficient for certain applications, A can be more efficient in others, as well as bring some extra benefits such as [index-based cache searching](https://github.com/eBay/go-ovn/blob/15eca9a4c0d78063b44b9cb844ab1c2d3347a226/ovnimp_orm.go#L303). If we go with A), the cache might have to store `interface{}`.

I'm not saying it's difficult to do it on top of this PR, just a thought to keep in mind so we can minimize the code duplication later on.

Reply:
A) is fine by me. You can update the cache to be `interface{}` in #31.
There is no reason to do it here as it would just result in extra type assertions.